File: inmopr-frontend/src/backend/userService.js
/*
 * Copyright (c) 2025 inmopr
 * Licensed under the MIT License. See LICENSE file in the project root for full license information.
 */

import {config, appFetch, setServiceToken, getServiceToken, removeServiceToken, setReauthenticationCallback} from './appFetch';

export const login = (email, password, onSuccess, onErrors, reauthenticationCallback) =>
    appFetch('/users/login', config('POST', {email, password}),
        authenticatedUser => {
            setServiceToken(authenticatedUser.serviceToken);
            setReauthenticationCallback(reauthenticationCallback);
            onSuccess(authenticatedUser);
        },
        onErrors);

export const logout = () => removeServiceToken();

export const signUp = (user, onSuccess, onErrors, reauthenticationCallback) => {
    appFetch('/users/signUp', config('POST', user),
        authenticatedUser => {
            setServiceToken(authenticatedUser.serviceToken);
            setReauthenticationCallback(reauthenticationCallback);
            onSuccess(authenticatedUser);
        },
        onErrors);
}

export const updateProfile = (user, onSuccess, onErrors) =>
    appFetch(`/users/${user.id}`, config('PUT', user),
        onSuccess, onErrors);

File: inmopr/src/main/java/com/tfm/inmopr/model/services/UserService.java
/*
 * Copyright (c) 2025 inmopr
 * Licensed under the MIT License. See LICENSE file in the project root for full license information.
 */

package com.tfm.inmopr.model.services;

import com.tfm.inmopr.model.entities.User;
import com.tfm.inmopr.model.exceptions.DuplicateInstanceException;
import com.tfm.inmopr.model.exceptions.IncorrectLoginException;
import com.tfm.inmopr.model.exceptions.InstanceNotFoundException;

import java.time.LocalDate;
import java.util.List;
import java.util.Optional;

public interface UserService {

    User login(String email, String password) throws IncorrectLoginException;

    void signUp(User user) throws DuplicateInstanceException;

    User updateProfile(Long id, String firstName, String lastName, String email, LocalDate birthDate, String password,
           List<String> favorites) throws DuplicateInstanceException, InstanceNotFoundException;

    Optional<User> findById(Long id);

}
